src/ipc/ipc4/handler.c

```diff
@@ -465,7 +465,7 @@ static int ipc4_process_chain_dma(struct ipc4_message_request *ipc4)
 	bool delay = false;
 	int ret;
 
-	memcpy_s(&cdma, sizeof(cdma), ipc4, sizeof(cdma));
+	memcpy_s(&cdma, sizeof(cdma), ipc4, sizeof(*ipc4));
```

Comment:
This is an unchecked memcpy_s call which _may_ fail and I believe in case of failure there's an issue (the destination gets zeroed out, so you need to pay attention to that to avoid bugs like this). If this returns an error you should propagate or assert it doesn't happen (using the non-removable SOF asserts). Ditto on all other memcpy_s calls in this patch.

Reply:
this hasn't been changed by this commit. This commit doesn't increase chances of  `memcpy_s()` to fail

Reply:
Can we take the memcpy_s check as a separate patch/PR. This needs to be fixed first.